Test that brew man changes no files

scpeters opened the pull request:
I believe that `brew man` is not part of CI. This runs it in the test-bot and verifies that it doesn't cause any changes to the tracked files. I put this very early in the test-bot loop to speed up testing, but we could put it elsewhere. I believe the master branch of brew currently will fail this test.

scpeters commented on cmd/brew-test-bot.rb:
```diff
@@ -710,6 +710,11 @@ def homebrew
+        test "brew", "man"
+        test "git", "diff", "--exit-code", "-C", HOMEBREW_REPOSITORY, "docs/brew.1.html"
+        test "git", "diff", "--exit-code", "-C", HOMEBREW_REPOSITORY, "manpages/brew.1"
+        test "git", "diff", "--exit-code", "-C", HOMEBREW_REPOSITORY, "manpages/brew-cask.1"
```
Actually, I guess I could just get the `git diff` of the `manpages` folder and save a line

MikeMcQuaid replied:
Maybe it's worth adding this inside `brew man` itself?

scpeters replied:
Interesting...call `git diff --exit-code` from `brew man` and return that exit code...that would be a nice place to put it.

scpeters replied:
added in https://github.com/Homebrew/brew/pull/1177, I'll update this PR accordingly...

MikeMcQuaid commented:
Great idea, was on my list to check this out 👍

scpeters commented:
I updated to use the `--fail-if-changed` option instead of `--exit-code`.

Also I added a plain `brew man` test to improve coverage. This extra call adds another 2 seconds or so to test-bot.

MikeMcQuaid commented on cmd/brew-test-bot.rb:
```diff
@@ -710,6 +710,11 @@ def homebrew
 
       if !@tap && (@formulae.empty? || @test_default_formula)
+        # this should always succeed
+        test "brew", "man"
```
Can remove the first call.

scpeters replied:
done
